opentelemetry-stdout/src/metrics/exporter.rs

```diff
@@ -76,57 +77,132 @@ impl PushMetricsExporter for MetricsExporter {
     }
 
     fn shutdown(&self) -> Result<()> {
-        self.writer.lock()?.take();
+        self.is_shutdown.store(true, atomic::Ordering::SeqCst);
         Ok(())
     }
 }
 
+fn print_metrics(metrics: &[ScopeMetrics]) {
+    for (i, metric) in metrics.iter().enumerate() {
+        println!("\tInstrumentation Scope #{}", i);
+        println!("\t\tName         : {}", &metric.scope.name);
+        if let Some(version) = &metric.scope.version {
+            println!("\t\tVersion  : {:?}", version);
+        }
+        if let Some(schema_url) = &metric.scope.schema_url {
+            println!("\t\tSchemaUrl: {:?}", schema_url);
+        }
+        metric
+            .scope
+            .attributes
+            .iter()
+            .enumerate()
+            .for_each(|(index, kv)| {
+                if index == 0 {
+                    println!("\t\tScope Attributes:");
+                }
+                println!("\t\t\t ->  {}: {}", kv.key, kv.value);
+            });
+
+        metric.metrics.iter().enumerate().for_each(|(i, metric)| {
+            println!("Metric #{}", i);
+            println!("\t\tName         : {}", &metric.name);
+            println!("\t\tDescription  : {}", &metric.description);
+            println!("\t\tUnit         : {}", &metric.unit);
+
+            let data = metric.data.as_any();
+            if let Some(hist) = data.downcast_ref::<data::Histogram<u64>>() {
+                println!("\t\tType         : Histogram");
+                print_histogram(hist);
+            } else if let Some(hist) = data.downcast_ref::<data::Histogram<f64>>() {
+                println!("\t\tType         : Histogram");
+                print_histogram(hist);
+            } else if let Some(_hist) = data.downcast_ref::<data::ExponentialHistogram<i64>>() {
```

Comment:
Similar to https://github.com/open-telemetry/opentelemetry-rust/pull/2040#discussion_r1727888812, I think we only support `u64` and `f64` for ExponentialHistograms.